dtls.c

```diff
@@ -914,7 +931,34 @@ static int verify_ext_ec_point_formats(uint8 *data, size_t data_length) {
   }
 
   dtls_warn("no supported ec_point_format found\n");
-  return dtls_alert_fatal_create(DTLS_ALERT_HANDSHAKE_FAILURE);
+  return 1;
+}
+
+static int verify_ext_sig_hash_algo(uint8 *data, size_t data_length) {
+  int i, hash_type, sig_type;
+
+  /* length of sig_hash_algo list */
+  i = dtls_uint16_to_int(data);
+  data += sizeof(uint16);
+  if (i + sizeof(uint16) != data_length) {
+    dtls_warn("the list of the supported signature_algorithms should be tls extension length - 2\n");
+    return 1;
+  }
+
+  for (i = data_length - sizeof(uint8); i > 0; i -= sizeof(uint8)) {
```

Comment:
This does not look right: We iterate data_length -1 times, and in each iteration, `data` is increased by two bytes, i.e., in the end, we have read 2 · (data_length - 1) bytes.

Reply:
Corrected to below.  Result of a copy and past and partial update - sorry.
````
  for (i = data_length - sizeof(uint16); i > 0; i -= sizeof(uint16)) {
````